overlord/snapstate/snapmgr.go

```diff
@@ -399,6 +399,37 @@ func (snapst *SnapState) IsComponentInCurrentSeq(cref naming.ComponentRef) bool
 	return snapst.Sequence.ComponentSideInfoForRev(idx, cref) != nil
 }
 
+// IsCurrentComponentRevInAnyNonCurrentSeq tells us if the component cref in
+// the revision for the current snap is used in another sequence point too.
+func (snapst *SnapState) IsCurrentComponentRevInAnyNonCurrentSeq(cref naming.ComponentRef) bool {
+	// Find current component in current sequence point
+	currentIdx := snapst.LastIndex(snapst.Current)
+	if currentIdx == -1 {
+		return false
+	}
+	currentSeqPt := snapst.Sequence.Revisions[currentIdx]
+	currentComp := currentSeqPt.FindComponent(cref)
+	if currentComp == nil {
+		return false
+	}
+
+	// Find if the current component revision is used elsewhere
+	for idx, seqPt := range snapst.Sequence.Revisions {
+		if idx == currentIdx {
+			continue
+		}
+		compInSeqPt := seqPt.FindComponent(cref)
+		if compInSeqPt == nil {
+			continue
+		}
+		if compInSeqPt.SideInfo.Revision == currentComp.SideInfo.Revision {
+			return true
+		}
+	}
```

Comment:
is some of it maybe a SnapSequence helper?

Reply:
I have moved most of this to the sequence package.